day-7.lisp

```diff
@@ -0,0 +1,55 @@
+(defpackage aoc-day-7
+  (:use :cl)
+  (:export :answers))
+(in-package :aoc-day-7)
+
+(defun parse-wire-instruction (circuit instruction)
+  (let ((parts (reverse (uiop:split-string instruction)))
+        wire-name)
+    (setq wire-name (car parts))
+    (setq parts (mapcar (lambda (s) (if (every 'digit-char-p s) (parse-integer s) s)) (reverse (cddr parts))))
+    (setf (gethash wire-name circuit) parts)))
+
+(defun eval-arg (circuit wires arg)
+  (if (integerp arg)
+      arg
+      (let ((v (gethash arg wires)))
+        (unless v
+          (setq v (eval-wire circuit wires arg))
+          (setf (gethash arg wires) v))
+        v)))
+
+(defun eval-wire (circuit wires wire-name)
+  (let ((wire (gethash wire-name circuit))
+        argc)
+    (setq argc (length wire))
+    (logand
+      65535
```

Comment:
doesn't look like common lisp has constants for stuff like this?

Reply:
Yeah looks like it. Updated to use hex notation.